src/raven/notifications_group.vala

```diff
@@ -135,18 +137,62 @@ namespace Budgie {
 				notification.destroy(); // Nuke the notification
 				update_count(); // Update our counter
 				dismissed_notification(id); // Notify anything listening
-
 				if (count == 0) { // This was the last notification
 					dismissed_group(app_name); // Dismiss the group
 				}
 			}
 		}
 
+		/**
+		 * too many notifications will choke raven and the desktop, so let's set a limit;
+		 * keep the latest n-notifications of current group, delete older ones
+		 */
+		public void limit_notifications () {
+
+			GLib.List<uint32> currnotifs = notifications.get_keys();
+			CompareFunc<int> intcmp = (a, b) => {
+				return (int) (a > b) - (int) (a < b);
+			};
+			currnotifs.sort(intcmp);
+			uint n_currnotifs = currnotifs.length();
+			/**
+			 * no need to reduce if the current number of notifications is below our threshold
+			 * and we shouldn't attempt to set a negative uint
+			 */
+			if (n_currnotifs <= tokeep) {
+				return;
+			}
+			uint n_remove = n_currnotifs - tokeep;
+			print(@"inside: count = $count, n_remove is $n_remove\n");
```

Comment:
```suggestion
			debug("inside: count = %d, n_remove is %d", count, n_remove);
```

Reply:
This was a line, just for checking if all went well, can be removed.